### PR TITLE
[ci] Remove runtime-benchmarks from tests

### PR DESCRIPTION
**Pull request by alvicsam**
cc https://github.com/paritytech/ci_cd/issues/864#issuecomment-1700891142

### CODE REVIEW

File: .gitlab/pipeline/test.yml [hunk 1]
```diff
@@ -45,6 +45,22 @@ test-linux-oldkernel-stable:
...
+    - .common-refs
+    - .run-immediately
+    - .pipeline-stopper-artifacts
+  variables:
+    RUST_TOOLCHAIN: stable
+    # Enable debug assertions since we are running optimized builds for testing
+    # but still want to have debug assertions.
+    RUSTFLAGS: "-Cdebug-assertions=y -Dwarnings"
+  script:
+    - cargo nextest run --features runtime-benchmarks benchmarking
```
Comment (ggwpez) [inline .gitlab/pipeline/test.yml:62]:
```suggestion
    - cargo nextest run --features runtime-benchmarks benchmarking --locked -q --profile testnet
```

**Review by ggwpez — APPROVED** [review]
I dont think this is actually going to run all benchmarking tests, but there is no good solution currently.

**Review by juangirini — APPROVED** [review]

**Comment by gilescope** [comment]
Ah cool didn't see this so created another: https://github.com/paritytech/polkadot-sdk/pull/1360 - snuk the extra benchmark tests into an existing job so little additional load on the CI. Also looks to me there's a few other locations we want to remove the feature from.

File: .gitlab/pipeline/test.yml (hunk 1 above)
Comment (gilescope) [reply .gitlab/pipeline/test.yml:62]: All the benchmark tests are under modules called `benchmark::` so if we remove the 'ing' off the end then we should catch all of the autogenerated ones.

**Review by chevdor — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-additional-tests
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3547106

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-additional-tests
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3548282

**Review by ggwpez — APPROVED** [review]
What is the status of this? I think i actually need it for another MR…

**Comment by alvicsam** [comment]
> What is the status of this? I think i actually need it for another MR…

I tried to move combine the benchmarks with another job but it takes too long. I moved it to a separate job and if pipeline succeeds we can merge it

File: .gitlab/pipeline/test.yml [hunk 2]
```diff
@@ -92,7 +92,24 @@ test-linux-stable-additional-tests:
         --locked \
         --release \
         --features runtime-benchmarks,try-runtime
-  allow_failure: true
+    # https://github.com/paritytech/ci_cd/issues/864
+    - time cargo nextest run --features runtime-benchmarks benchmark --locked --cargo-profile testnet
```
Comment (altaua) [inline .gitlab/pipeline/test.yml:96]: Why is this run both here and below in the new job?

Comment (alvicsam) [reply .gitlab/pipeline/test.yml]: Nice catch, forgot to remove the duplicate, thanks

**Review by altaua — APPROVED** [review]

**Review by gilescope — APPROVED** [review]

**Pull request merged**